typing/typecore.ml

```diff
@@ -1188,18 +1303,24 @@ and type_pat_aux ~exception_allowed ~constrs ~labels ~no_existentials ~mode
         with Not_found -> None
       in
       let constr =
-        match lid.txt, constrs with
-          Longident.Lident s, Some constrs ->
-            assert (Hashtbl.mem constrs s); Hashtbl.find constrs s
+        let synthetic_constrs = match mode with
+          | Normal -> None
+          | Counter_example {synthetic_names = {constrs; _}; _} ->
+             Some constrs
+        in
+        match lid.txt, synthetic_constrs with
+        | Longident.Lident s, Some constrs ->
+            assert (Hashtbl.mem constrs s);
```

Comment:
Is it excessive if I ask that you add `(* cf {!counter_example_synthetic_names} *)` on this assert?

Reply:
done